miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultLibraryTemplateService.java

```diff
@@ -0,0 +1,124 @@
+package uk.ac.bbsrc.tgac.miso.service.impl;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Consumer;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import uk.ac.bbsrc.tgac.miso.core.data.impl.DetailedLibraryTemplate;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.LibraryTemplate;
+import uk.ac.bbsrc.tgac.miso.core.store.DeletionStore;
+import uk.ac.bbsrc.tgac.miso.core.store.LibraryTemplateStore;
+import uk.ac.bbsrc.tgac.miso.core.store.SecurityProfileStore;
+import uk.ac.bbsrc.tgac.miso.core.util.PaginationFilter;
+import uk.ac.bbsrc.tgac.miso.service.LibraryTemplateService;
+import uk.ac.bbsrc.tgac.miso.service.security.AuthorizationManager;
+
+@Transactional(rollbackFor = Exception.class)
+@Service
+public class DefaultLibraryTemplateService implements LibraryTemplateService {
```

Comment:
All public methods should filter or throw auth exceptions based on the project's SecurityProfile